kubernetes/resource_kubernetes_network_policy.go

```diff
@@ -0,0 +1,359 @@
+}
+
+func resourceKubernetesNetworkPolicyCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*kubernetes.Clientset)
+
+	metadata := expandMetadata(d.Get("metadata").([]interface{}))
+	spec, err := expandNetworkPolicySpec(d.Get("spec").([]interface{}))
+	if err != nil {
+		return err
+	}
+
+	svc := api.NetworkPolicy{
+		ObjectMeta: metadata,
+		Spec:       *spec,
+	}
+	log.Printf("[INFO] Creating new network policy: %#v", svc)
+	out, err := conn.NetworkingV1().NetworkPolicies(metadata.Namespace).Create(&svc)
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[INFO] Submitted new network policy: %#v", out)
+	d.SetId(buildId(out.ObjectMeta))
+
+	return resourceKubernetesNetworkPolicyRead(d, meta)
+}
+
+func resourceKubernetesNetworkPolicyRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*kubernetes.Clientset)
+
+	namespace, name, err := idParts(d.Id())
+	if err != nil {
+		return err
+	}
+	log.Printf("[INFO] Reading network policy %s", name)
+	svc, err := conn.NetworkingV1().NetworkPolicies(namespace).Get(name, meta_v1.GetOptions{})
+	if err != nil {
+		log.Printf("[DEBUG] Received error: %#v", err)
+		return err
+	}
+	log.Printf("[INFO] Received network policy: %#v", svc)
+	err = d.Set("metadata", flattenMetadata(svc.ObjectMeta))
+	if err != nil {
+		return err
+	}
+
+	flattened := flattenNetworkPolicySpec(svc.Spec)
+	log.Printf("[DEBUG] Flattened network policy spec: %#v", flattened)
+	err = d.Set("spec", flattened)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func resourceKubernetesNetworkPolicyUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*kubernetes.Clientset)
+
+	namespace, name, err := idParts(d.Id())
+	if err != nil {
+		return err
+	}
+
+	ops := patchMetadata("metadata.0.", "/metadata/", d)
+	if d.HasChange("spec") {
+		diffOps, err := patchNetworkPolicySpec("spec.0.", "/spec", d)
+		if err != nil {
+			return err
+		}
+		ops = append(ops, *diffOps...)
+	}
+	data, err := ops.MarshalJSON()
+	if err != nil {
+		return fmt.Errorf("Failed to marshal update operations: %s", err)
+	}
+	log.Printf("[INFO] Updating network policy %q: %v", name, string(data))
+	out, err := conn.NetworkingV1().NetworkPolicies(namespace).Patch(name, pkgApi.JSONPatchType, data)
+	if err != nil {
+		return fmt.Errorf("Failed to update network policy: %s", err)
+	}
+	log.Printf("[INFO] Submitted updated network policy: %#v", out)
+	d.SetId(buildId(out.ObjectMeta))
+
+	return resourceKubernetesNetworkPolicyRead(d, meta)
+}
+
+func resourceKubernetesNetworkPolicyDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*kubernetes.Clientset)
+
+	namespace, name, err := idParts(d.Id())
+	if err != nil {
+		return err
+	}
+	log.Printf("[INFO] Deleting network policy: %#v", name)
+	err = conn.NetworkingV1().NetworkPolicies(namespace).Delete(name, &meta_v1.DeleteOptions{})
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[INFO] Network Policy %s deleted", name)
+
+	d.SetId("")
```

Comment:
This isn't actually needed. Terraform will do that automatically on delete. (I know some other resources do this - they need a cleanup).

Reply:
Done.